### PR TITLE
Feature: add/refactor  query capabilities for `Asset`

### LINKED ISSUE #1459: Feature: add query capabilities for Asset

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

This PR slightly refactors the SQL-AssetIndex in the following regards:
- introduces a test targeting an actual Postgres DB
- changes the `dataAddress` column to datatype `json`
- introduces Postgres-specific dialect 
- moves the dynamic query assembly into the statements class

## Why it does that

The `AssetIndex` already had the capability to dynamically assemble the SQL query, so it was refactored cleaned up a bit.

## Further notes

Due to the special nature how Asset properties are stored, using the `*Mapping` classes was not necessary.

## Linked Issue(s)

Closes #1459 

## Checklist

- [x] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [x] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: extensions/sql/asset-index-sql/docs/schema.sql [hunk 1]
```diff
@@ -14,29 +14,32 @@
 -- THIS SCHEMA HAS BEEN WRITTEN AND TESTED ONLY FOR POSTGRES
 
 -- table: edc_asset
-CREATE TABLE IF NOT EXISTS edc_asset (
-    asset_id VARCHAR NOT NULL,
-    PRIMARY KEY (asset_id)
-    );
+CREATE TABLE IF NOT EXISTS edc_asset
+(
+    id VARCHAR NOT NULL,
```
Comment (ndr-brt) [inline extensions/sql/asset-index-sql/docs/schema.sql:19]: Some other sql scripts uses the `<entityname>_id` convention, I guess this is needed to permit mapping with jdbc, but wouldn't be better to have the `asset_id` field in the property table called `asset_id_fk`?

File: extensions/sql/asset-index-sql/src/test/java/org/eclipse/dataspaceconnector/sql/assetindex/schema/postgres/PostgresAssetIndexTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    private static final String POSTGRES_DATABASE = "itest";
+    private static final String JDBC_URL_PREFIX = "jdbc:postgresql://localhost:5432/";
+    private SqlAssetIndex sqlAssetIndex;
+    private BaseSqlDialectStatements sqlStatements;
+    private TransactionContext transactionContext;
+    private Connection connection;
+
+    @BeforeAll
+    static void prepare() {
+        // todo: reuse org.eclipse.dataspaceconnector.test.e2e.postgresql.PostgresqlLocalInstance??
```
Comment (ndr-brt) [inline extensions/sql/asset-index-sql/src/test/java/org/eclipse/dataspaceconnector/sql/assetindex/schema/postgres/PostgresAssetIndexTest.java:74]: In my opinion: yes, let's reuse it :)

File: extensions/sql/asset-index-sql/src/test/java/org/eclipse/dataspaceconnector/sql/assetindex/schema/postgres/PostgresAssetIndexTest.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+            executeQuery(connection, "DROP TABLE " + dialect.getAssetPropertyTable() + " CASCADE");
+        });
+        doCallRealMethod().when(connection).close();
+        connection.close();
+    }
+
+    @Test
+    void query_byAssetProperty() {
+        var allAssets = IntStream.range(0, 5).mapToObj(i -> {
+            var asset = createAssetBuilder("test-asset" + i)
```
Comment (ndr-brt) [inline extensions/sql/asset-index-sql/src/test/java/org/eclipse/dataspaceconnector/sql/assetindex/schema/postgres/PostgresAssetIndexTest.java:136]: this function that maps an integer to an integer to an asset and insert it into the asset index could be extracted as it is used 3 times (not in the same way, but very similarly)

File: extensions/sql/asset-index-sql/docs/schema.sql (hunk 1 above)
Comment (paullatzelsperger) [reply extensions/sql/asset-index-sql/docs/schema.sql:19]: ok, so the PK would be `asset_id`, and the FK would be `asset_id_fk`

Comment (ndr-brt) [reply extensions/sql/asset-index-sql/docs/schema.sql]: yes, sorry for the bad grammar :smiling_face_with_tear:

**Review by ndr-brt — APPROVED** [review]

**Pull request merged**